crates/dojo-core/src/world_factory.cairo

```diff
@@ -0,0 +1,107 @@
+use array::ArrayTrait;
+
+#[contract]
+mod WorldFactory {
+    use dojo_core::interfaces::IWorldDispatcher;
+    use dojo_core::interfaces::IWorldDispatcherTrait;
+    use starknet::ContractAddress;
+    use starknet::ClassHash;
+    use starknet::syscalls::deploy_syscall;
+    use array::ArrayTrait;
+    use traits::Into;
+    use option::OptionTrait;
+    use starknet::contract_address::ContractAddressIntoFelt252;
+
+    struct Storage {
+        world_class_hash: ClassHash,
+        executor_address: ContractAddress,
+    }
+
+    #[event]
+    fn WorldSpawned(address: ContractAddress) {}
+
+    #[constructor]
+    fn constructor(world_class_hash_: ClassHash, executor_address_: ContractAddress) {
+        world_class_hash::write(world_class_hash_);
+        executor_address::write(executor_address_);
+    }
+
+    #[external]
+    fn spawn(name: felt252, components: Array::<ClassHash>, systems: Array::<ClassHash>) {
+        // deploy world
+        let mut world_constructor_calldata: Array<felt252> = ArrayTrait::new();
+        world_constructor_calldata.append(name);
+        world_constructor_calldata.append(executor_address::read().into());
+        let world_class_hash = world_class_hash::read();
+        let result = deploy_syscall(world_class_hash, 0, world_constructor_calldata.span(), true);
+        let (world_address, _) = result.unwrap_syscall();
+
+        // events
+        WorldSpawned(world_address);
+
+        // register components
+        let components_len = components.len();
+        register_components(components, components_len, 0_usize, world_address);
+
+        // register systems
+        let systems_len = systems.len();
+        register_systems(systems, systems_len, 0_usize, world_address);
+    }
+
+    #[external]
+    fn set_executor(executor_address_: ContractAddress) {
```

Comment:
We need to add access control (only owner is prob ok for now) but can do so in a followup PR